test/test.ts

```diff
@@ -1490,6 +1490,12 @@ test('is.nonEmptyMap', t => {
 	});
 });
 
+test('is.propertyKey', t => {
+	testType(t, 'string', ['emptyString', 'numericString']);
+	testType(t, 'number', ['nan', 'integer', 'safeInteger', 'infinite']);
+	testType(t, 'symbol');
```

Comment:
@PopGoesTheWza

Looking at other tests, it seems that `testType` is to be used like the following:

```ts
test('is.XXX', t => {
	testType(t, 'XXX', [...exceptions]);
});
```

So given that, you should:

1. Add `propertyKey` to the `types` map (line 43 currently)
2. Update the test to the following:

```ts
test('is.propertyKey', t => {
	testType(t, 'propertyKey', [...exceptions]);
});
```

Where `exceptions` is probably:

```ts
[
	'string', 'emptyString', 'numericString', 
	'number', 'nan', 'integer', 'safeInteger', 'infinite', 
	'symbol',
]
```

Which is the merge of whatever is tests would also end up positive for your predicate.

Reply:
@gioragutt it won't work. Assuming I create a new `Test`:

```ts
	['propertyKey', {
		is: is.propertyKey,
		assert: assert.propertyKey,
		fixtures: [
			undefined
		],
		typename: 'propertyKey'
	}],
```

1. Attribute `typename` needs to be a value from `type TypeName = ObjectTypeName | PrimitiveTypeName`. Neither of the two include `propertyKey` and adding it would bee semantically pointless (IMHO)
2. Attribute `typename` is used to filter entries from the `types` map (i.e. entries `number`, `nan`, `integer`, `safeInteger` and `infinite` all  share the same `typename`.) but it only support a single value. In this case, `typename` would need to support multiple values (i.e. `['string', 'number', 'symbol']`) in order to benefit from existing entries.
3. Modifying `testType()` in order to support many `typename`s is a bit complex and a task I would not venture without good reason

Reply:
Oh, got it. Makes sense, guess I forgot the code. `testType` should be used to, well, test a *Type*.

Since `PropertyKey` isn't a type, but a union of types, `testType` does not fit.

However, it should still have tests that look something like:

```ts
test('is.propertyKey', t => {
	t.true(is.propertyKey('key'));
	t.true(is.propertyKey(5));
	t.true(is.propertyKey(Symbol('')));

	t.false(is.propertyKey(true));
	t.false(is.propertyKey({}));
	t.false(is.propertyKey([]));
	t.false(is.propertyKey(new Map()));
	t.false(is.propertyKey(new Set()));
});
```

If you see another case for either `true` cases or `false` cases feel free to add them.